### ISSUE TITLE
About loss in model and answer loss

### ISSUE THREAD

**Issue by yeppp27** [cliangyu/Cola#3]
Hi! Thanks for your great work! Can I ask about the difference and effect of loss in model and answer loss in finetune.py? Thanks in adavance

**Comment by cliangyu** [comment]
We attempted to assign more loss to the answer part of the language model output by adjusting args.answer_factor.
It didn't turn out to work. Therefore it was not reported in the paper.

**Comment by yeppp27** [comment]
Thanks for your kind reply! And I also want to ask the difference of next token prediction loss and teacher forcing loss!  There are some sentences of ``To optimize the LLM, we use the language modeling loss for next-token prediction, with the teacher forcing strategy. " in your paper. Looking forward to your reply!

**Comment by yeppp27** [comment]
By the way! Will mutli-choices VQA data set training perform better than direct answer VQA data set training?

**Comment by cliangyu** [comment]
Language models are autoregressive transformers, i.e., their input is their own last output. When we have a ground truth answer, teacher forcing at training is forcefully feeding the correct outputs instead of feeding its own garbage to it. 
We are also using the next prediction loss. They are orthogonal.

**Comment by cliangyu** [comment]
Mutli-choices VQA is always easier than direct answer VQA. This is intuitive.
Also, I think the evaluation protocols of direct answer VQA datasets used in our paper are outdated: they try to match predictions with a set of answers verbatim. The latest VQA datasets usually use language model based evaluation, which is more appropriate.

**Comment by yeppp27** [comment]
Thanks!
